src/lib/homestore_backend/tests/hs_blob_tests.cpp

```diff
@@ -72,6 +72,67 @@ class HomeObjectFixture : public ::testing::Test {
         }
     }
 
+    using blob_map_t = std::map< std::tuple< pg_id_t, shard_id_t, blob_id_t >, homeobject::Blob >;
+
+    void put_blob(blob_map_t& blob_map, std::vector< std::pair< pg_id_t, shard_id_t > > const& pg_shard_id_vec,
+                  uint64_t const num_blobs_per_shard, uint32_t const max_blob_size) {
+        std::uniform_int_distribution< uint32_t > rand_blob_size{1u, max_blob_size};
+        std::uniform_int_distribution< uint32_t > rand_user_key_size{1u, 1 * 1024};
+
+        for (const auto& id : pg_shard_id_vec) {
+            int64_t pg_id = id.first, shard_id = id.second;
+            for (uint64_t k = 0; k < num_blobs_per_shard; k++) {
+                uint32_t alignment = 512;
+                // Create non 512 byte aligned address to create copy.
+                if (k % 2 == 0) alignment = 256;
+
+                std::string user_key;
+                user_key.resize(rand_user_key_size(rnd_engine));
+                BitsGenerator::gen_random_bits(user_key.size(), (uint8_t*)user_key.data());
+                auto blob_size = rand_blob_size(rnd_engine);
+                homeobject::Blob put_blob{sisl::io_blob_safe(blob_size, alignment), user_key, 42ul};
+                BitsGenerator::gen_random_bits(put_blob.body);
+                // Keep a copy of random payload to verify later.
+                homeobject::Blob clone{sisl::io_blob_safe(blob_size, alignment), user_key, 42ul};
+                std::memcpy(clone.body.bytes, put_blob.body.bytes, put_blob.body.size);
+                auto b = _obj_inst->blob_manager()->put(shard_id, std::move(put_blob)).get();
+                ASSERT_TRUE(!!b);
+                auto blob_id = b.value();
+
+                LOGINFO("Put blob pg {} shard {} blob {} data {}", pg_id, shard_id, blob_id,
+                        hex_bytes(clone.body.bytes, std::min(10u, clone.body.size)));
+                blob_map.insert({{pg_id, shard_id, blob_id}, std::move(clone)});
+            }
+        }
+    }
+
+    void verify_get_blob(blob_map_t const& blob_map, bool const use_random_offset = false) {
+        uint32_t off = 0, len = 0;
+        for (const auto& [id, blob] : blob_map) {
+            int64_t pg_id = std::get< 0 >(id), shard_id = std::get< 1 >(id), blob_id = std::get< 2 >(id);
+            len = blob.body.size;
+            if (use_random_offset) {
+                std::uniform_int_distribution< uint32_t > rand_off_gen{0u, blob.body.size - 1u};
```

Comment:
Where do we need the check? If the `use_random_offset` flag is set we generate a value to the len, otherwise we don't use the len.

Reply:
yes for the `rand_len_gen` you set it to [1, blob_size] which avoid the len to be zero.
I am asking,  by our api definition len=0 means read till end of the blob,   are we on-purpose skipping this case or just overlook?

Reply:
ohh yeah, I just modularized the existing code into functions to re-use the code. I am not sure why len is generated from 1 to blob_size.
I think it is because the other case (reading till the end of the blob) is already covered above. (use_random_offset = false case)